lychee-lib/src/types/input.rs

```diff
@@ -132,7 +133,8 @@ impl Input {
     ) -> Result<Self> {
         let source = if value == STDIN {
             InputSource::Stdin
-        } else if let Ok(url) = Url::parse(value) {
+        } else if let (Ok(_), Ok(url)) = (Request::builder().uri(value).body(()), Url::parse(value))
```

Comment:
Your version from the issue description was a bit more readable to me, but that's probably just personal taste. Both works, so it's up to you to decide if you like to change it back.

Apart from that, can we add a comment here why this is needed? Essentially the main gist from the issue to let people know that it's a workaround for Windows.

Reply:
I switched to this because the other version caused some tests to break on Linux. Apparently there are cases that RequestBuilder catches but Url doesn't, and vice versa.

Reply:
That makes sense. It would also sounds like an important fact to document here.